Actions/CheckForUpdates/CheckForUpdates.ps1

```diff
@@ -372,171 +170,94 @@ try {
                     # file exists, compare and add to $updateFiles if different
                     $dstContent = Get-ContentLF -Path $dstFile
                     if ($dstContent -cne $srcContent) {
-                        Write-Host "Updated $name ($(Join-Path $dstPath $filename)) available"
+                        Write-Host "Updated $type ($(Join-Path $dstPath $filename)) available"
                         $updateFiles += @{ "DstFile" = Join-Path $dstPath $filename; "content" = $srcContent }
                     }
                 }
                 else {
                     # new file, add to $updateFiles
-                    Write-Host "New $name ($(Join-Path $dstPath $filename)) available"
+                    Write-Host "New $type ($(Join-Path $dstPath $filename)) available"
                     $updateFiles += @{ "DstFile" = Join-Path $dstPath $filename; "content" = $srcContent }
                 }
             }
         }
+        finally {
+            Pop-Location
+        }
     }
+}
 
-    if (-not $update) {
-        # $update not set, just issue a warning in the CI/CD workflow that updates are available
-        if (($updateFiles) -or ($removeFiles)) {
-            OutputWarning -message "There are updates for your AL-Go system, run 'Update AL-Go System Files' workflow to download the latest version of AL-Go."
-            AddTelemetryProperty -telemetryScope $telemetryScope -key "updatesExists" -value $true
-        }
-        else {
-            Write-Host "No updates available for AL-Go for GitHub."
-            AddTelemetryProperty -telemetryScope $telemetryScope -key "updatesExists" -value $false
-        }
+if (-not $update) {
+    # $update not set, just issue a warning in the CI/CD workflow that updates are available
+    if (($updateFiles) -or ($removeFiles)) {
```

Comment:
With the new `templateSha` setting, wouldn't a simple check be enough to determine if updates are available?
Like:
if the `templateSha` is not set => updates are available
if `templateSha` is set, but is different from the latest SHA on the branch from the template URL => updates are available
otherwise => updates are not available

Also, out of curiosity, `templateSha` is the same SHA we use for referencing the actions in the workflows, isn't it?
Like, `microsoft/AL-Go-Actions/ReadSettings@<templateSha>`

Reply:
If project depth has changed or other settings that affects workflow files - then updates would be available without the sha changing.

On the templateSha, it would be the Sha of microsoft/AL-Go-PTE or of fork/AL-Go. So, I guess sometimes, it would be the same - but not always.